swift/Sources/FlatBuffers/ByteBuffer.swift

```diff
@@ -352,7 +372,14 @@ public struct ByteBuffer {
   ///   - position: the index of the object in the buffer
   @inline(__always)
   public func read<T>(def: T.Type, position: Int) -> T {
-    _storage.memory.advanced(by: position).load(as: T.self)
+    #if swift(>=5.7)
+    if allowReadingUnalignedBuffers {
+      return _storage.memory.advanced(by: position).loadUnaligned(as: T.self)
+    }
```

Comment:
Maybe just wrap this if check with the directive. Then we don't have to duplicate L379 and L382.

Reply:
If possible, can you provide an example since I honestly couldn't figure it out.

Reply:
My bad. I was thinking something like this

```swift
public func read<T>(def: T.Type, position: Int) -> T {
    #if swift(>=5.7)
    if allowReadingUnalignedBuffers {
      return _storage.memory.advanced(by: position).loadUnaligned(as: T.self)
    }
    #endif
    return _storage.memory.advanced(by: position).load(as: T.self)
}
```

Reply:
Thanks! and fixed